src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,119 @@
 /* eslint-disable */
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getTodos } from '../../api';
+import { todosSlice } from '../../features/todos';
+import cn from 'classnames';
+import { currentTodoSlice } from '../../features/currentTodo';
+
+function filterTodos(todos: any, query: string, status: string) {
+  let filteredTodos = todos;
+  if (query) {
+    filteredTodos = todos.filter((todo: any) =>
+      todo.title.toLowerCase().includes(query.toLowerCase()),
+    );
+  }
+  if (status !== 'all') {
+    filteredTodos = filteredTodos.filter(
+      (todo: any) => todo.completed === (status === 'completed'),
+    );
+  }
+  return filteredTodos;
+}
```

Comment:
You can move this to another file. 
Also, you need to write an enum for all statuses.